src/services/post-service.ts

```diff
@@ -0,0 +1,13 @@
+import { Post } from "../shared/types";
+
+export default class postService {
+  getAll(): Post[] {
```

Comment:
Both of your services has the getAll method maybe you can create an abstract class with generic types as we saw the last class.